lib/ansible/module_utils/k8s/raw.py

```diff
@@ -93,12 +98,13 @@ def __init__(self, *args, **kwargs):
             if LooseVersion(self.openshift_version) < LooseVersion("0.8.0"):
                 self.fail_json(msg="openshift >= 0.8.0 is required for validate")
         self.append_hash = self.params.get('append_hash')
-        if self.append_hash:
-            if not HAS_K8S_CONFIG_HASH:
-                self.fail_json(msg="openshift >= 0.7.2 is required for append_hash")
+        if self.append_hash and not HAS_K8S_CONFIG_HASH:
+            self.fail_json(msg="openshift >= 0.7.2 is required for append_hash")
         if self.params['merge_type']:
```

Comment:
One thing to note, merge_type might not work with apply, we might hard code the apply merge strategy to merge (since strategic merge patch would need to know the merge key to intelligently generate the patch)

Reply:
Yes, this is true, I was thinking that merge_type and apply should be mutually exclusive, I just clearly forgot to actually do that!

Reply:
Did we still want to do this?